### PR TITLE
[ML][Pipelines] Support serverless compute in pipeline & add end-to-end tests

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

- bypass getting compute from service when it's serverless
- add end-to-end tests, and skip in playback mode for now

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/tests/pipeline_job/e2etests/test_pipeline_job.py [hunk 1]
```diff
@@ -1696,6 +1696,22 @@ def check_name_version_and_register_succeed(output, asset_name):
         assert pipeline_job.jobs["compare"].outputs.best_model.name == "best_model"
         assert pipeline_job.jobs["compare"].outputs.best_model.version == random_version
 
+    @pytest.mark.skipif(condition=not is_live(), reason="component version changes across tests")
```
Comment (D-W-) [inline sdk/ml/azure-ai-ml/tests/pipeline_job/e2etests/test_pipeline_job.py:1699]: Add a task number to track this, can directly use the existing task. 2177353 and update these case to it

Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/tests/pipeline_job/e2etests/test_pipeline_job.py]: Updated the skip reason with this work item.

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/tests/test_configs/pipeline_jobs/serverless_compute/all_types/automl/pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+$schema: https://azuremlschemas.azureedge.net/latest/pipelineJob.schema.json
+type: pipeline
+display_name: Serverless in Pipeline - AutoML
+settings:
+  default_compute: azureml:serverless
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/test_configs/pipeline_jobs/serverless_compute/all_types/automl/pipeline.yml:5]: looks like we only do test on serverless compute in pipeline settings?

File: sdk/ml/azure-ai-ml/tests/pipeline_job/e2etests/test_pipeline_job.py [hunk 3]
```diff
@@ -1900,6 +1900,22 @@ def check_name_version_and_register_succeed(output, asset_name):
         assert pipeline_job.jobs["compare"].outputs.best_model.name == "best_model"
         assert pipeline_job.jobs["compare"].outputs.best_model.version == random_version
 
+    @pytest.mark.skipif(condition=not is_live(), reason="Task 2177353: component version changes across tests.")
+    @pytest.mark.parametrize(
+        "yaml_path",
+        [
+            "./tests/test_configs/pipeline_jobs/serverless_compute/all_types/command/pipeline.yml",
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/pipeline_job/e2etests/test_pipeline_job.py:1907]: Maybe re-org like this:
./tests/test_configs/pipeline_jobs/all_types/command/pipeline_serverless_compute.yml

I think we might add more such cross component type pipeline feature

File: sdk/ml/azure-ai-ml/tests/test_configs/pipeline_jobs/serverless_compute/all_types/automl/pipeline.yml (hunk 2 above)
Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/tests/test_configs/pipeline_jobs/serverless_compute/all_types/automl/pipeline.yml:5]: Currently yes, will add some node-level test on this in next PR.

File: sdk/ml/azure-ai-ml/tests/pipeline_job/e2etests/test_pipeline_job.py (hunk 3 above)
Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/tests/pipeline_job/e2etests/test_pipeline_job.py:1907]: Sure, will do this in next PR.

**Pull request merged**